Util.h

```diff
@@ -0,0 +1,10 @@
+#pragma once
+
+std::vector<std::string> SplitStringByCharacter(std::string string, char splitter);
+bool FileExistsInEnvironmentPath(std::string filename);
+std::string GetExecutableFullFilename();
+std::string GetExecutableDirectory();
+std::string GetWorkingDirectory();
+bool FileExistsInExecutableDirectory(std::string filename);
+bool FileExistsInWorkingDirectory(std::string filename);
+bool CanFindFileInEnvironment(std::string filename);
```

Comment:
Try to always use const strings as arguments, much nicer on copying

Reply:
const std::string& val_

Reply:
Good catch - I will fix this.
Let me know if you notice any others. I'm not great at remembering my const correctness.

Reply:
Fixed